cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -593,8 +614,13 @@ protected ContextDefinitions readContextDefinitions(String path) throws Exceptio
         
         List<CqlEvaluationRequest> requestsForContext = requests.getEvaluationsForContext(contextName);
         
-        Map<String, Object> expressionResults = new HashMap<>();
+        // parameters json -> {columnName, result}
+        Map<String,Map<String, Object>> expressionResultsByParameters = new HashMap<>();
         for (CqlEvaluationRequest request : requestsForContext) {
+            
+            String parametersJson = CqlEvaluationRequestUtil.getKeyParametersColumnData(request, getKeyParameterNames());
```

Comment:
I'm not a big fan of creating a JSON string from a POJO for every request for every context value.
If I'm not mistaken, the resulting JSON will not differ between context values.
I'd like to see some way of pre-computing the strings once in the driver and sent to the executors either via serialization (i.e. as a property in `SparkCqlEvaluator`) or broadcast variables.
Caching the serialized form in a `transient` variable somewhere would work as well.

If you can confidently say that this results in a negligible performance loss overall, then I'm OK with leaving it.

Reply:
I added a cache for the calculated parameters JSON.